DndEncounterSimulator/Objects/Creature.py

```diff
@@ -112,3 +124,28 @@ def find_opponent(self, fighters: List[Creature]) -> Union[None, int]:
             if fighter.camp != self.camp:
                 return index
         return None
+
+    def find_best_weapon(self) -> int:
+        """
+        Method to find the best weapon (ie deals statistically the more damage)
+        #TODO: add options and limitations for two-handed, ranged, damage immunity...
+        :return: (int) the index of the best weapon to use
+        """
+        best = 0  # we assume we have at least one weapon
+        best_damage = self.weapons[best].damage
+        best_sum = int(best_damage.split("d")[0]) * int(best_damage.split("d")[1])
+        for (index, weapon) in enumerate(self.weapons):
+            tmp_damage = weapon.damage
+            tmp_sum = int(tmp_damage.split("d")[0]) * int(tmp_damage.split("d")[1])

```

Comment:
I'd recommend to avoid naming your variables with abbreviations, it makes code harder to understand and to maintain.

```suggestion
            temporary_damage = weapon.damage
            temporary_sum = int(temporary_damage.split("d")[0]) * int(temporary_damage.split("d")[1])

```